backend/matching-service/spec/matching.spec.js

```diff
@@ -0,0 +1,113 @@
+process.env.JASMINE = true;
+
+const request = require('supertest');
+const app = require('../index.js');
+
+describe('enqueue function', () => {
+  let match1;
+  let match2;
+
+  beforeAll(() => {
```

Comment:
Might want to ensure a queue is empty before each test case as well, to prevent side-effects.

Also, as mentioned in the PR description, which tests are failing as a result of the nature of the service?

Reply:
Its the "should enqueue a user" test that's failing, its primarily because of the way we implemented the request having no response until a match is found or it times out. At least I think that's the most likely reason for failure, however it passes like a vast majority of the time. We could leave out this matching test case since we already have unit tests for the queue itself?

Reply:
Updated with a timeout, works fine now.